### PR TITLE
Correct type hints per PEP 484

### PR DESCRIPTION
**Pull request by mrbcmorris**
## Description
Solves #2747 by ensuring all type annotations are behind a `# type:` comment.

## Checklist
- [ ] Added to the correct milestone.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Library documentation is updated.
- [ ] [Corp site](https://github.com/DataDog/documentation/) documentation is updated (link to the PR).

### CODE REVIEW

**Review by brettlangdon — COMMENTED** [review]
great catch!

can you add a release note for this?

```
pip install reno
reno new some-name-for-the-note
```

**Comment by mrbcmorris** [comment]
> great catch!
> 
> can you add a release note for this?
> 
> ```
> pip install reno
> reno new some-name-for-the-note
> ```

Sure thing.  I'm not positive how much detail is warranted here so I've left it vague at the moment.  Let me know if you have any guidance on what detail it might ought contain.

File: ddtrace/tracer.py [hunk 1]
```diff
@@ -234,7 +234,7 @@ async def web_handler(request):
         return ctx
 
     def current_trace_context(self, *args, **kwargs):
-        # type (...) -> Optional[Context]
+        # type: (...) -> Optional[Context]
         """Return the context for the current trace.
 
         If there is no active trace then None is returned.
```
Comment (brettlangdon) [inline ddtrace/tracer.py:240]: I cannot add the suggestion on the line, but we need to change L242

```diff
-        active = self.context_provider.active(*args, **kwargs)
+        active = self.context_provider.active()
```

Comment (mrbcmorris) [reply ddtrace/tracer.py:240]: resolved in 7263d36c011583afa7347339d4078752ee0aaf27

File: releasenotes/notes/correct-type-annotations-2ee30779966960b6.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    Corrected some typing anntotations for PEP 484 compliance
```
Comment (brettlangdon) [inline releasenotes/notes/correct-type-annotations-2ee30779966960b6.yaml:4]:
```suggestion
    Corrected some typing annotations for PEP 484 compliance
```

https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/10474/workflows/365f7ed0-f811-4c75-b5e9-6d8a43e91972/jobs/800999?invite=true#step-105-349

Comment (mrbcmorris) [reply releasenotes/notes/correct-type-annotations-2ee30779966960b6.yaml]: resolved in 275acdc86921bc28b9925e629f3ec4eb2865fcee

**Review by brettlangdon — APPROVED** [review]

**Comment by brettlangdon** [comment]
@Mergifyio backport 0.51

**Review by majorgreys — APPROVED** [review]

**Comment by brettlangdon** [comment]
thanks @mrbcmorris !

once tests pass this should auto-merge. I'll keep an eye on the tests and can re-run if there are any flaky ones.

we'll also backport this to a bugfix release on 0.51

**Pull request merged**